source/postprocess/visualization/dynamic_topography.cc

```diff
@@ -28,19 +28,60 @@ namespace aspect
     namespace VisualizationPostprocessors
     {
       template <int dim>
-      std::pair<std::string, Vector<float> *>
-      DynamicTopography<dim>::execute() const
+      DynamicTopography<dim>::
+      DynamicTopography ()
+        :
+        DataPostprocessorScalar<dim> ("dynamic_topography",
+                                      update_quadrature_points)
+      {}
+
+      template <int dim>
+      void
+      DynamicTopography<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
       {
+        for (unsigned int q=0; q<computed_quantities.size(); ++q)
+          computed_quantities[q](0) = 0;
+
         const Postprocess::DynamicTopography<dim> &dynamic_topography =
           this->get_postprocess_manager().template get_matching_postprocessor<Postprocess::DynamicTopography<dim> >();
 
-        std::pair<std::string, Vector<float> *>
-        return_value ("dynamic_topography",
-                      new Vector<float>(dynamic_topography.cellwise_topography()));
+        auto cell = input_data.template get_cell<DoFHandler<dim> >();
+
+        bool cell_at_top_or_bottom_boundary = false;
```

Comment:
Maybe the code could use some more comment to make reading it easier. You could for example put a comment here like: "We are only interested in the faces at the top and bottom of the domain, so we check that here."